packages/block-editor/src/components/block-toolbar/unified.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { Popover } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import BlockContextualToolbar from '../block-list/block-contextual-toolbar';
+
+export default function UnifiedBlockToolbar() {
+	const hasFixedToolbar = useSelect(
+		( select ) => select( blockEditorStore ).getSettings().hasFixedToolbar,
+		[]
+	);
+
+	if ( hasFixedToolbar ) {
+		return (
+			<>
+				<Popover.Slot name="block-toolbar" />
```

Comment:
Do we still need this slot? I have trouble understanding why this `UnifiedBlockToolbar` doesn't directly render `BlockContextualToolbar` in all  cases?

Reply:
Because the block popover needs access to the block nodes, which is only available in the context of the BlockList. I plan to change this in the future though, so then we'll be able to render it here in place.

Reply:
The important thing is that the slot becomes just an internal thing.

Reply:
Once we have #29573, this will be possible.

Reply:
Rebase and moved the insertion point and popover toolbar into the unified component.